### PR TITLE
Add handle change password

### PR DESCRIPTION
**Pull request by TanPingZhi**
LGTM
closes #79

### CODE REVIEW

File: user-service/controller/user-controller.js [hunk 1]
```diff
@@ -37,6 +37,61 @@ export async function createUser(req, res) {
...
+// The first check is done in the frontend
+const isPasswordComplex = (password) => {
+  const minLength = 8;
+  const hasUpperCase = /[A-Z]/.test(password);
+  const hasSpecialChar = /[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/.test(
+    password
+  );
+
+  return password.length >= minLength && hasUpperCase && hasSpecialChar;
+};
```
Comment (wr1159) [inline user-service/controller/user-controller.js:51]: hmm is it common practice to check in backend also? i see the point but wondering if it's normal

File: frontend/components/user-settings/user-settings.tsx [hunk 2]
```diff
@@ -223,7 +223,54 @@ export default function UserSettings({ userId }: { userId: string }) {
...
+          headers: {
+            Authorization: `Bearer ${token}`,
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            currentPassword,
+            newPassword,
+          }),
+        }
+      );
```
Comment (wr1159) [inline frontend/components/user-settings/user-settings.tsx:248]: maybe can abstract it out

**Review by wr1159 — APPROVED** [review]
LGTM, just some questions

File: frontend/components/user-settings/user-settings.tsx (hunk 2 above)
Comment (TanPingZhi) [reply frontend/components/user-settings/user-settings.tsx:248]: will make new pr for that

**Pull request merged**